oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/RoutingUtils.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.elasticsearch.base;
+
+import org.apache.skywalking.oap.server.core.storage.model.Model;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class RoutingUtils {
+
+    public static Optional<String> getRouting(final Model model, final Map<String, Object> builder) {
+        Optional<String> routingField = model.getElasticSearchModelExtension().getRouting();
+        return routingField.map(v -> extractRoutingValue(v, builder));
```

Comment:
What does this `map` mean? `Optional` -> `map` -> `Optional` again?

Reply:
```
if `routingField` is empty:
    just return emtpy
else
    apply process func, either empty or some(value)
```

common patterns of functional programming

Reply:
I know what is `map` function. I think what you are trying to process?

Reply:
check if the value of routingFiled is presented.

Reply:
OK, I think I am following the logic now.

But this is actually wrong. Check `ElasticSearchConverter#ToStorage#accept`. 

Your change actually works because this only works as the entity don't merge. Trace segment and Zipkin span are not merging with others, but the codes should not be written in this way.

You should ensure `IndexRequest` includes this `routingValue`(Optional). You can't rely on the builder's key as column name as always.